kong/plugins/oauth2/access.lua

```diff
@@ -172,9 +178,17 @@ local function authorize(conf)
   -- Adding the state if it exists. If the state == nil then it won't be added
   response_params.state = state
 
+  -- Appending kong generated params to redirect_uri query string
+  if parsed_redirect_uri then
+    if not parsed_redirect_uri.query then
+      parsed_redirect_uri.query = ""
+    end
+    parsed_redirect_uri.query = ngx.encode_args(utils.table_merge(ngx.decode_args(parsed_redirect_uri.query), response_params))
```

Comment:
Maybe better to use `utils.encode_args` as it preserves percent-encoding of the querystring. See #749